pallets/oracle-feed/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Oracle pallet to feed values.
+//! Feeding is permissionless given an initial fee for each key.
+//! This pallet do not aggregate/validate any, it just store them by account as
+//! they come. It's expected other pallet read this storage and
+//! aggregate/validate the values.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use cfg_traits::{fees::PayFee, ValueProvider};
+	use frame_support::{pallet_prelude::*, traits::Time};
+	use frame_system::pallet_prelude::*;
+
+	use crate::weights::WeightInfo;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	type MomentOf<T> = <<T as Config>::Time as Time>::Moment;
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Identify an oracle value
+		type OracleKey: Parameter + Member + Copy + MaxEncodedLen;
+
+		/// Represent an oracle value
+		type OracleValue: Parameter + Member + Copy + MaxEncodedLen + Default;
+
+		/// A way to obtain the current time
+		type Time: Time;
+
+		/// Fee for the first time a feeder feeds a value
+		type FirstValuePayFee: PayFee<Self::AccountId>;
+
+		/// The weight information for this pallet extrinsics.
+		type WeightInfo: WeightInfo;
+	}
+
+	/// Store all oracle values indexed by feeder
+	#[pallet::storage]
+	pub(crate) type FedValues<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::AccountId,
```

Comment:
This should not be `T::AccountId` but rather an `T::Origin`. This allows us to have democracy provide oracles too. Either by council or by `Origin::Root` if wanted to.